tests/src/test/java/edu/snu/onyx/tests/runtime/master/scheduler/FaultToleranceTest.java

```diff
@@ -181,11 +168,42 @@ public void testContainerRemoval() throws Exception {
     final IREdge e4 = new IREdge(DataCommunicationPatternProperty.Value.OneToOne, v4, v5, Coder.DUMMY_CODER);
     irDAGBuilder.connectVertices(e4);
 
-    final DAG<IRVertex, IREdge> irDAG = CompiletimeOptimizer.optimize(irDAGBuilder.buildWithoutSourceSinkCheck(),
-        new TestPolicy(), "");
-    final DAG<PhysicalStage, PhysicalStageEdge> physicalDAG = irDAG.convert(physicalPlanGenerator);
+    DAG<IRVertex, IREdge> irDAG;
+    DAG<PhysicalStage, PhysicalStageEdge> physicalDAG = null;
+    try {
+      irDAG = CompiletimeOptimizer.optimize(irDAGBuilder.buildWithoutSourceSinkCheck(),
+          new TestPolicy(), "");
+      physicalDAG = irDAG.convert(physicalPlanGenerator);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return new PhysicalPlan("TestPlan", physicalDAG, physicalPlanGenerator.getTaskIRVertexMap());
+  }
 
-    final PhysicalPlan plan = new PhysicalPlan("TestPlan", physicalDAG, physicalPlanGenerator.getTaskIRVertexMap());
+  /**
+   * Tests fault tolerance after a container removal.
+   */
+  @Test(timeout=10000)
+  public void testContainerRemoval() throws Exception {
```

Comment:
Can you add a multi-threaded test that fails with the previous code, and passes with this PR?

Reply:
testTaskGroupReexecutionForFailure() fails with the previous code (occasionally) and passes with this PR.